### PR TITLE
Hcal backward segmentation r

### PR DESCRIPTION
**Pull request by lkosarz**
### Briefly, what does this PR introduce?

Update of backward HCal geometry to shrink it to the segmented sensitive volume.

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [x] New feature (issue #232)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### CODE REVIEW

File: compact/hcal/hcal_backward.xml [hunk 1]
```diff
@@ -16,6 +16,16 @@
...
+
+  <limits>
+  </limits>
+
+  <regions>
+  </regions>
+
+  <display>
+  </display>
+
```
Comment (wdconinc) [inline compact/hcal/hcal_backward.xml:28]: Don't introduce empty environments.
```suggestion
```

File: compact/hcal/hcal_backward.xml [hunk 2]
```diff
@@ -59,4 +69,7 @@
     </readout>
   </readouts>
 
-</lccdd>
+  <plugins>
+  </plugins>
```
Comment (wdconinc) [inline compact/hcal/hcal_backward.xml:73]:
```suggestion
```

File: compact/hcal/hcal_backward.xml [hunk 3]
```diff
@@ -59,4 +69,7 @@
...
+
+</lccdd>
```
Comment (wdconinc) [inline compact/hcal/hcal_backward.xml:75]: End file with newline.

File: compact/definitions.xml [hunk 4]
```diff
@@ -606,7 +605,7 @@ Service gaps in FW direction (before endcapP ECAL) and BW direction (before endc
     <constant name="HcalEndcapN_length"   value="44.0*cm"/>
     <constant name="HcalEndcapN_zmax"     value="HcalEndcapN_zmin + HcalEndcapN_length"/>
     <comment> HcalEndcapN needs to clear the straight beampipe with 1 * crossing angle + its radius of 22.25 mm </comment>
-    <constant name="HcalEndcapN_rmin"     value="max((HcalEndcapN_zmin + HcalEndcapN_length) * tan(abs(CrossingAngle)) + 22.25 * mm, 11 * cm)"/>
+    <constant name="HcalEndcapN_rmin"     value="19.9431*cm"/>
```
Comment (wdconinc) [inline compact/definitions.xml:608]: These are not hard-coded values that should be in the definitions file. These are implementation details only relevant to the HcalEndcapN. The definitions file should be like the detector parameter table. This is not a number in the detector parameter table, so it should not be in the definitions file.

Same with rmax.

**Comment by lkosarz** [comment]
Thank you! I made the suggested changes.

**Comment by wdconinc** [comment]
Well,... We do want to keep the rmin and rmax as they are defined by the detector parameter table in the definitions file. You can use a new variable in the hcal endcap compact file that is always greater than the detector parameter table rmin, and a new rmax that is always smaller than the rmax from the detector parameter table.

**Comment by lkosarz** [comment]
I understand. I separated the definitions between the compact file and parameter table.

File: compact/definitions.xml [hunk 5]
```diff
@@ -615,7 +614,7 @@ Service gaps in FW direction (before endcapP ECAL) and BW direction (before endc
     <constant name="HcalBarrelBackward_zmax" value="BackwardServiceGap_zmin"/>
     <constant name="HcalBarrel_length"    value="HcalBarrelForward_zmax + HcalBarrelBackward_zmax"/>
     <constant name="HcalBarrel_offset"    value="(HcalBarrelForward_zmax - HcalBarrelBackward_zmax)/2"/>
-
+
```
Comment (wdconinc) [inline compact/definitions.xml:617]:
```suggestion

```

**Review by wdconinc — APPROVED** [review]

**Review by wdconinc — DISMISSED** [review]

File: compact/hcal/backward.xml [hunk 6]
```diff
@@ -32,8 +36,8 @@
       <dimensions
         numsides="HcalEndcapN_CaloSides"
         zmin="HcalEndcapN_zmin"
-        rmin="HcalEndcapN_rmin"
-        rmax="HcalEndcapN_rmax"/>
+        rmin=HcalEndcapN_segments_rmin
+        rmax="HcalEndcapN_segments_rmax"/>
```
Comment (wdconinc) [inline compact/hcal/backward.xml:40]:
```suggestion
        rmin="HcalEndcapN_segments_rmin"
        rmax="HcalEndcapN_segments_rmax"/>
```

**Review by wdconinc — APPROVED** [review]

**Comment by lkosarz** [comment]
I committed a fix to the segmentation error. I did it by extending the segmentation to the corners. I'm unable to test it now, but will ask my student to do it ASAP. Sorry for trouble!

**Comment by wdconinc** [comment]
I don't see a PR. Let me know when it is ready.

**Pull request merged**